mollie-payments-for-woocommerce/includes/mollie/wc/helper/data.php

```diff
@@ -308,25 +308,25 @@ public function getRecurringPaymentMethods( $test_mode = false, $use_cache = tru
 		return self::$recurring_api_methods;
 	}
 
-	protected function getApiPaymentMethods( $test_mode = false, $use_cache = true, $filters = array () ) {
+	public function getApiPaymentMethods( $test_mode = false, $use_cache = true, $filters = array () ) {
 
-		$methods = array ();
+		$methods = false;
 
-		try {
-
-			$filters_key  = ( ! empty ( $filters['sequenceType'] ) ) ? '_' . $filters['sequenceType'] : '';
-			$transient_id = Mollie_WC_Plugin::getDataHelper()->getTransientId( 'api_methods_' . ( $test_mode ? 'test' : 'live' ) . $filters_key );
+		$filters_key = $filters;
+		$filters_key['test'] = ( $test_mode ? 'test' : 'live' );
+		$filters_key['api'] = 'methods';
 
-			if ( $use_cache ) {
-				$cached_methods = unserialize( get_transient( $transient_id ) );
+		try {
 
-				if ( $cached_methods && is_array($cached_methods) ) {
+			$transient_id = Mollie_WC_Plugin::getDataHelper()->getTransientId( md5(http_build_query($filters_key))  );
```

Comment:
Yes, I think `http_build_query` is the easiest way to create a string from an array. `serialize` can be disabled on some environments.